keystore/v2/keystore/api/tests/key.go

```diff
@@ -190,13 +193,12 @@ func testKeyInvalidInputs(t *testing.T, newKeyStore NewKeyStore) {
 		ValidUntil: time.Now().Add(time.Hour),
 		Data: []api.KeyData{
 			api.KeyData{
-				Format:     api.ThemisPublicKeyFormat,
-				PrivateKey: []byte("ompf"),
+				Format: api.ThemisKeyPairFormat,
 			},
 		},
 	})
 	if err != api.ErrNoKeyData {
-		t.Errorf("cannot add public key without data: %v", err)
+		t.Errorf("cannot add key pair without any data: %v", err)
```

Comment:
> we can add key without any data for cases when key material removed but we store metadata for history, yes?

Not really. This test verifies that it is not allowed to add *only* key metadata, without any key material. The reasoning is that you should not be backdating metadata alone.

However, it is possible (or, well, should be possible) to add a key with metadata then at some point destroy the key material so that only metadata remains.

Though, it’s an open question how this should interact with key export/import. I don’t know whether we should allow exporting destroyed keys and importing them back (which will be metadata-only import).